File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.java
Solution in java 

class Solution {
    public int findFinalValue(int[] nums, int original){
        // step 1. defined a data structured.
        Set<Integer> set = new HashSet<>();
        // step 2 . store the elements into it
        for(int i = 0; i < nums.length; i++){
            set.add(nums[i]);
        }
        // step 3 we keep on multiply while the set contains the numbers
        while(set.contains(original)){
            original = original * 2;
        }
        return original;
    }
}

solution in c++
    

File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.py
class Solution:
    def findFinalValue(self, nums: list[int], original: int) -> int:
        # Step 1: Convert list to set for fast lookup
        s = set()
        for i in range(len(nums)):
            s.add(nums[i])
        
        # Step 2: Keep multiplying while original exists in the set
        while original in s:
            original *= 2
        
        # Step 3: Return final value
        return original
